regression/snapshot-harness/structs_02/main.c

```diff
@@ -0,0 +1,41 @@
+#include <assert.h>
+
+struct S
+{
+  int c1;
+  int c2;
+} st;
+
+int *p;
+
+void initialize()
+{
+  st.c1 = 1;
+  st.c2 = 3;
+  p = &(st.c2);
+}
+
+void checkpoint()
+{
+}
+
+int foo()
+{
+  initialize();
+  checkpoint();
+
+  assert(st.c1 + 2 == st.c2);
+  assert(st.c1 + 2 == *p);
+  assert(*p == st.c2);
+  *p = st.c2 + 1;
+  assert(*p == st.c2);
+
+  return 0;
+}
+
+int main()
+{
+  foo();
+
+  return 0;
+}
```

Comment:
We need a different, smaller regression test. Maybe one in which the source location given to `--initial-source-location` is an empty line in the C program, and thus does not appear in the corresponding goto program.

Reply:
Added.